### PR TITLE
ParaSwap `/swap` API

### PR DESCRIPTION
**Pull request by squadgazzz**
### Description
ParaSwap provides a private `/swap` API in order to send a single API call instead of two(`/prices`+`/transactions`).
This should reduce the requests amount and also simplify the code.

### How to test
Local run

### CODE REVIEW

File: src/infra/dex/paraswap/dto.rs [hunk 1]
```diff
@@ -157,64 +99,36 @@ pub enum Side {
...
-            token_transfer_proxy: parsed.token_transfer_proxy,
-        })
-    }
-}
-
 #[serde_as]
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct Transaction {
+pub struct TxParams {
```
Comment (Mateo-mro) [inline src/infra/dex/paraswap/dto.rs:131]: why the change of name?

File: src/infra/dex/paraswap/mod.rs [hunk 2]
```diff
@@ -50,73 +50,41 @@ impl ParaSwap {
...
         &self,
         order: &dex::Order,
-        slippage: &dex::Slippage,
+        slippage: dex::Slippage,
         tokens: &auction::Tokens,
     ) -> Result<dex::Swap, Error> {
-        let price = self.price(order, tokens).await?;
-        let transaction = self.transaction(&price, order, tokens, slippage).await?;
+        let query = dto::SwapQuery::new(&self.config, order, tokens, slippage)?;
+        let swap = util::http::roundtrip!(
```
Comment (Mateo-mro) [inline src/infra/dex/paraswap/mod.rs:61]: why not to make it into a funcion as `price()` was?

File: src/infra/dex/paraswap/dto.rs (hunk 1 above)
Comment (squadgazzz) [reply src/infra/dex/paraswap/dto.rs:131]: Because the field is called `txParam`

File: src/infra/dex/paraswap/mod.rs (hunk 2 above)
Comment (squadgazzz) [reply src/infra/dex/paraswap/mod.rs:61]: Do you mean dto into domain? Because it also requires an `order` instance for the domain object to be constructed.

Comment (squadgazzz) [reply src/infra/dex/paraswap/mod.rs:61]: I think price and transaction functions were separated because they called different logical endpoints. We now have a single endpoint and I don't think it makes sense to extract it since it will be used just in a single place and doesn't have a lot of lines of code.

File: src/infra/dex/paraswap/mod.rs [hunk 3]
```diff
@@ -50,73 +50,41 @@ impl ParaSwap {
...
+            <dto::Swap, dto::Error>;
+            self.client.request(reqwest::Method::GET, util::url::join(&self.config.endpoint, "swap"))
+                .header("X-API-KEY", &self.config.api_key)
```
Comment (MartinquaXD) [inline src/infra/dex/paraswap/mod.rs:64]: Setting the header is no longer needed with https://github.com/gnosis/solvers/pull/32, right?

File: src/infra/dex/paraswap/dto.rs [hunk 4]
```diff
@@ -52,13 +51,20 @@ pub struct PriceQuery {
...
+    pub slippage: u16,
 }
 
-impl PriceQuery {
+impl SwapQuery {
     pub fn new(
         config: &super::Config,
         order: &dex::Order,
         tokens: &auction::Tokens,
+        slippage: dex::Slippage,
```
Comment (MartinquaXD) [inline src/infra/dex/paraswap/dto.rs:67]: This could be taken by reference which would also get rid of a bunch of other changes need to have this available as a value here.

**Review by MartinquaXD — APPROVED** [review]
Looks good, just 2 nits.
Really excited for this change! Maybe the paraswap quoter finally become useful. 😅

**Review by Mateo-mro — APPROVED** [review]

File: src/infra/dex/paraswap/dto.rs (hunk 4 above)
Comment (squadgazzz) [reply src/infra/dex/paraswap/dto.rs:67]: https://github.com/gnosis/solvers/blob/f05a36eec420ed534d925e2fa85031b874413c10/src/infra/dex/paraswap/mod.rs#L114 requires it to be copied then.

Comment (MartinquaXD) [reply src/infra/dex/paraswap/dto.rs:67]: I think cloning this is fine to keep the API more consistent across different dexes. 👌
